PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitHelper.java

```diff
@@ -173,6 +183,8 @@ public void printRunningLogStreamingly(Project project, int id, IClusterDetail c
         } catch (Exception e) {
             if (HDInsightUtil.getSparkSubmissionToolWindowManager(project).getJobStatusManager().isJobKilled() == false) {
                 HDInsightUtil.getSparkSubmissionToolWindowManager(project).setError("Error : Failed to getting running log. Exception : " + e.toString());
+
+                postEventProperty.put("SubmitFailedReason", e.toString().substring(0, 50));
```

Comment:
We limit the string value length to 50 everywhere. It might be better to implement a special TelementryEventProperty class for this.